File: packages/rules/test/fixtures/square/js/missing-signature-verification.positive.ts
import express from "express";

const app = express();
app.use(express.json());

app.post("/webhooks/square", async (req, res) => {
  // BUG: no HMAC verification.
  const event = req.body;
  console.log("Square event:", event.event_id);
  res.status(200).end();
});

app.listen(3000);

File: packages/rules/test/fixtures/square/py/missing-signature-verification.positive.py
from flask import Flask, request

app = Flask(__name__)


@app.route("/webhooks/square", methods=["POST"])
def square_webhook():
    # BUG: no HMAC verification.
    event = request.get_json()
    print("Square event:", event.get("event_id"))
    return "", 200
